src/container/WeatherContainer/WeatherContainer.tsx

```diff
@@ -55,12 +72,41 @@ export const WeatherContainer: React.FC = () => {
     };
   };
 
+  const CityInput = () => {
+    return (
+      <div className="chooser">
+        <input
+          value={userDefinedCity}
+          onChange={checkRegexExpression}
+          placeholder="Your City"
+        />
+        <Button onClick={onButtonClick}>Send</Button>
+      </div>);
+  };
+
+  const checkRegexExpression = (inputElement: React.ChangeEvent<HTMLInputElement>) => {
+    const value: string = inputElement.target.value;
+
+    if (!value) {
+      setUserDefinedCity('');
+    } else if (userInputRegex.test(value)) {
+      setUserDefinedCity(value);
+    }
+  };
+
+  const onButtonClick = () => {
+    userDefinedCity && fetchWeatherDataWithUserDefinedCity(userDefinedCity)
+      .then(result => setWeatherData(prepareWeatherData(result)))
+      .then(() => setWeatherDataReady(true));
```

Comment:
do we need this additional `then` here? ;x 

I mean.. cannot we move this `setWeatherDataReady(true)` line below `setWeatherData(prepareWeatherData(result))` in same `then` scope? ;x

Reply:
Done